analyzer/src/main/kotlin/managers/utils/MavenSupport.kt

```diff
@@ -790,7 +790,8 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) {
             sourceArtifact = sourceRemoteArtifact,
             vcs = vcsFromPackage,
             vcsProcessed = vcsProcessed,
-            isMetadataOnly = mavenProject.packaging == "pom" || isSpringMetadataProject,
+            isMetadataOnly = (mavenProject.packaging == "pom" && binaryRemoteArtifact.url.endsWith(".pom"))
```

Comment:
Is the pom URL always used as the binary artifact in these cases or do we also have to check if `binaryRemoteArtifact.url.isEmpty()`?

Reply:
There is indeed *one* case in `pub-expected-output-with-flutter-android-and-cocoapods.yml` where `is_metadata_only: true` and the binary artifact URL is empty. However, the URL shouldn't be empty (that's a bug in `Gradle` that `GradleInspector` fixes). So, phrased differently, in all cases (of our expected results) where *rightfully* `is_metadata_only: true`, the URL is non-empty and ends in `.pom`.